Qtof parameters Update

Ahlammentag opened the pull request:
In this PR, I modified the Qtof_preprocessing.qmd file based on the improvements we made in FTMS preprocessing. I added the CleanPeaksParam, also using the threshold based on three times the 75th percentile of the peak widths. I also included the beta-core-based selection.

For the centWave step, I used the same parameters as in FTMS, except that I did not change the peakwidth, since it is specific to the data:
```
qtof <- findChromPeaks(
  qtof,
  CentWaveParam(
    peakwidth = c(8, 20), integrate = 2, snthresh = 6,
    ppm = 50, prefilter = c(3, 500), mzdiff = 0.001
  ),
  chunkSize = 2
)
```
For MergeNeighboringPeaksParam, I changed the expandMz parameter to 0.01 instead of 0.001 (as used in FTMS):

`mnpp <- MergeNeighboringPeaksParam(expandRt = 10, expandMz = 0.01, minProp = 0.66)
`
For PeakDensityParam, I used two different configurations.
First:
```
pdp <- PeakDensityParam(
  sampleGroups = sampleData(ftms)$sample_id,
  bw = 6, minFraction = 0.3, binSize = 0.1, ppm = 10
)
```
Second (keeping bw = 2):

```
pdp <- PeakDensityParam(
  sampleGroups = sampleData(qtof)$sample_id,
  bw = 2, minFraction = 0.3
)

```
                        
@jorainer and @rebdau , If everything is fine with this preprocessing modifications, I can move to qtof filtering and replace in feature spectra the qtof object with dropFilledChromPeaks(qtof) ?

Ahlammentag committed:
qtof

Ahlammentag committed:
Merge remote-tracking branch 'origin/jomain' into ahlam

Ahlammentag committed:
Merge remote-tracking branch 'origin/jomain' into ahlam

Ahlammentag committed:
Remove raw-data-evaluation.qmd

Ahlammentag committed:
update qtof_filtering.qmd file

Ahlammentag committed:
add gaussian shape

rebdau commented on qtof-preprocessing.qmd:
````diff
@@ -383,21 +314,27 @@ quantile(unname(pks[, "mzmax"] - pks[, "mzmin"]))
 ```{r}
-mnpp <- MergeNeighboringPeaksParam(expandRt = 6, expandMz = 0.01)
+mnpp <- MergeNeighboringPeaksParam(expandRt = 10, expandMz = 0.01,
+                                   minProp = 0.66)
````
for ftms: `CentWaveParam(peakwidth = c(20, 80)); MergeNeighboringPeaksParam(expandRt = 10)`
for qtof: `CentWaveParam(peakwidth = c(8, 20)); MergeNeighboringPeaksParam(expandRt = 10)`
Proportionally, expandRt is much larger for qtof.
But we set expandRt with respect to the **observed peak widths**. Need to run the code and look at the actual distribution to decide if `expandRt = 10` is exagerated here.

jorainer replied:
We usually set the `expandRt` to ~ half of the observed average peak width. All other settings are fine I think.

Ahlammentag replied:
<img width="506" height="127" alt="image" src="https://github.com/user-attachments/assets/704a8c31-e0b7-4e08-b03f-1dc86beefbf0" />

Thanks @rebdau  and @jorainer ,above the peak width of qtof data, In this case maybe we should reduce the expandRt from 10 to 7, what do you think?

jorainer replied:
yes, 7 seconds sounds good to me!

rebdau committed:
Remove .Rhistory and ignore it

Ahlammentag committed:
small changes

Ahlammentag committed:
Save .Rhistory changes

Ahlammentag committed:
Resolve merge conflict and update qtof_filtering.qmd

Ahlammentag committed:
update qtof analysis

Ahlammentag committed:
qtof_filtering

Ahlammentag committed:
Updates

Ahlammentag commented:
Hi @rebdau and @jorainer, I updated the qtof_preprocessing.qmd file.
 Then, I used dropFilledChromPeaks with featureSpectra() like in FTMS, but we still had 13 duplicated assignments.
In the qtof_filtering.qmd file (you can find these files on my branch), I added some plots and visualizations for these special cases. I also implemented the code to remove duplicates based on differences in mzmed and rt. Now the results look good — we no longer have any duplicates.

Finally, I continued the process to select the best representative MS2 spectrum per feature.

Ahlammentag committed:
update qtof_filtering

Ahlammentag committed:
qtof_filterin

jorainer committed:
Update/refine preprocessing settings

Based on issue #20 this changes the preprocessing to use:
- `ppm = 25` for `CentWaveParam`
- `expandRt = 2`, `expandMz = 0.001` and `minProp = 0.75` for
`MergeNeighboringPeaksParam`
- `binSize = 0.1` and `ppm = 10` also for th second `PeakDensityParam` (used for
the final correspondence analysis).

jorainer committed:
Merge branch 'ahlam' into jomain

Ahlammentag committed:
Merge pull request #21 from rebdau/jomain

Refine preprocessing settings

Commit:
centroiding from RDynLib_ftms as template for centoriding MS/MS

Ahlammentag committed:
small correction

Ahlammentag committed:
qtof_filterin

Commit:
Merge branch 'ahlam' into rebdau
"ahlam updated preprocessing and filtering"

Ahlammentag committed:
convert_spectra_to_mgf

Commit:
settings for centroiding and precursor purity

Ahlammentag committed:
convert feature matrix format to a long table

Ahlammentag committed:
convert feature matrix format to a long table

Commit:
correct centroiding and precursor purity, add combinespectra

Commit:
correct combineSpectra and keep top intensities

Commit:
optimized parameters in qtof filtering

Commit:
small change introduction

Ahlammentag committed:
Merge pull request #24 from rebdau/rebdau

Rebdau

Ahlammentag committed:
qtof

Ahlammentag committed:
qtof

Ahlammentag committed:
small changes

Ahlammentag committed:
qtof_filtering

Ahlammentag committed:
qtof flax to sql

Ahlammentag committed:
qtof flax spectra to sql and compdb

jorainer committed:
Determine number of peaks along the analysis

Ahlammentag committed:
Merge pull request #25 from rebdau/jomain

Determine number of peaks along the analysis

Ahlammentag committed:
qtof_filtering

Ahlammentag committed:
qtof_filtering

Ahlammentag committed:
qtof neg to sql

Ahlammentag committed:
qtof flax to sql

Ahlammentag committed:
Flax qtofneg to sql

Ahlammentag committed:
Flax qtof to sql

Ahlammentag committed:
Flax qtof neg to sql

Ahlammentag committed:
qtof_filtering.qmd

Ahlammentag committed:
qtofneg_flax_sql.qmd

jorainer reviewed (approved):
Thanks @Ahlammentag ! I will merge and then check and clean the workflow locally.
